pkg/backend/impl/redis/redis.go

```diff
@@ -79,6 +80,20 @@ func (s *redis) Init(ctx context.Context) error {
 			MinVersion:         tls.VersionTLS12,
 			InsecureSkipVerify: s.args.TLSSkipVerify,
 		}
+
+		if s.args.CACertificate != "" {
+			tlscfg.InsecureSkipVerify = false
+			s.logger.Info("Adding CA Cert to TLS Config")
+			roots := x509.NewCertPool()
+			var (
+				certPEM = []byte(s.args.CACertificate)
+			)
+			ok := roots.AppendCertsFromPEM(certPEM)
+			if !ok {
+				return fmt.Errorf("could not add CA Certificate: %w", errors.New("Invalid CA Cert format"))
+			}
```

Comment:
InsecureSkipVerify should be false when the CA is informed. If a user specifies incompatible parameters we should validate and throw an error (I'll add a comment on this), so that first line can be removed.

There is a slightly more idiomatic way of doing those [conditionals in go using a local variable to the if block](https://go.dev/doc/effective_go#if).

```go
if ok := ... ; !ok {
  ...
}
```

The returned error already contains the stacktrace, there is no need to wrap an error inside of another here. I left the internal one that you already created ([non capitalised](https://github.com/golang/go/wiki/CodeReviewComments#error-strings))

```suggestion
			roots := x509.NewCertPool()
			if ok := roots.AppendCertsFromPEM([]byte(s.args.TLSCACertificate)); !ok {
				return errors.New("not valid CA Cert format")
			}
```

Reply:
Thanks, looks much cleaner this way.